src/Instrumentation/Guzzle/tests/Integration/GuzzleInstrumentationTest.php

```diff
@@ -165,4 +166,47 @@ public function test_headers_propagation(): void
         $this->client->get('/');
 
     }
+
+    /**
+     * @dataProvider exceptionProvider
+     */
+    public function test_exceptions_enabled_sets_response_attributes($response, ?int $expected = null): void
+    {
+        $client = new Client([
+            'handler' => $this->handlerStack,
+            'base_uri' => 'https://example.com/',
+            'http_errors' => true,
+            'exceptions' => true,
+        ]);
+        $this->mock->append($response);
+        $this->assertCount(0, $this->storage);
+
+        try {
+            $client->send(new Request('GET', 'https://example.com/error'));
+        } catch (\Exception $e) {
+            // Expected exception
```

Comment:
i wonder if we could `expectException` here

Reply:
I think not, because it would bail out if we didn't catch, and not run all the code after this point.